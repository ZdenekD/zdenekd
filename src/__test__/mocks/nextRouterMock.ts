const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
