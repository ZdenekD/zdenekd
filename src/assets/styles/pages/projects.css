.root {
    --section-background: hsl(var(--color-green-lighten-5));
    --selection-background: hsl(var(--color-green));
    --curtain-color: hsl(var(--color-green-lighten-4));
    --content-heading: hsl(var(--color-green));
    --content-heading-admit: hsl(var(--color-green-lighten-1));
    --key-background: hsl(var(--color-green-lighten-3));
}

.article {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
}
