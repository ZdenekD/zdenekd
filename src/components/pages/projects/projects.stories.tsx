import Projects from '.';

export default {
    title: 'Pages/Projects',
    parameters: {chromatic: {delay: 5000}},
};

export const component: React.FC = () => (
    <Projects />
);
