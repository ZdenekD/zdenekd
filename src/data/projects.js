const projects = {
    tomton: {
        id: 'tomton',
        title: 'Tomton',
        url: 'https://www.tomton.cz',
        description: '',
        tools: [
            'webpack',
            'react',
            'wordpress',
            'lottie',
            'anime',
        ],
    },
    stenclova: {
        id: 'stenclova',
        title: 'Ivana Štenclová',
        url: 'http://www.stenclova.cz',
        description: '',
        tools: [],
    },
    dirtjob: {
        id: 'dirtjob',
        title: 'Dirtjob',
        url: 'http://dirtjob.cz/',
        description: '',
        tools: [],
    },
    digitalvision: {
        id: 'digitalvision',
        title: 'DigitalVision',
        url: 'https://digitalvision.cz/',
        description: '',
        tools: [],
    },
    fitsmarky: {
        id: 'fitsmarky',
        title: 'Fit s Marky',
        url: 'https://fit-s-marky.cz/',
        description: '',
        tools: [],
    },
    portfolium: {
        id: 'portfolium',
        title: 'Portfolium',
        url: '',
        description: '',
        tools: [],
    },
    avprague: {
        id: 'avprague',
        title: 'AVPrague',
        url: '',
        description: '',
        tools: [],
    },
};

export default projects;
