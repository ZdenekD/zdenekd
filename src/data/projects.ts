import {LocalesEnum} from '@/enums';
import type {IProjects} from '@/types/projects';

const projects: IProjects = [
    {
        id: 'tomton-interior',
        title: 'Tomton Interiér',
        url: 'https://www.tomton-interier.cz',
        tools: [
            'vercel',
            'nextjs',
            'hygraph',
            'react',
            'typescript',
            'framer-motion',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Společnost specializující se na kompletní služby při zařizování komerčních i soukromých interiérů.'},
            [LocalesEnum.en]: {description: 'Company focused on complete services in furnishing commercial and private interiors.'},
        },
    },
    {
        id: 'tomton',
        title: 'Tomton',
        url: 'https://www.tomton.cz',
        tools: [
            'vercel',
            'nextjs',
            'hygraph',
            'react',
            'typescript',
            'lottie',
            'framer-motion',
            'animejs',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Společnost vyrabějící designové otopné systémy. Jeden z modelů získal celosvětově uznávané ocenění Red Dot Design Award.'},
            [LocalesEnum.en]: {description: 'Company producing design heating systems. One of the models won the world-renowned Red Dot Design Award.'},
        },
    },
    {
        id: 'jf-woodpall',
        title: 'JF Woodpall',
        url: 'https://www.jf-woodpall.cz',
        tools: [
            'vercel',
            'nextjs',
            'react',
            'typescript',
            'lottie',
            'framer-motion',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Společnost zaměřená na prodej, výrobu a opravu dřevěných palet.'},
            [LocalesEnum.en]: {description: 'Company focused on the sale, manufacture and repair of wooden pallets.'},
        },
    },
    {
        id: 'fit-s-marky',
        title: 'Fit s Marky',
        url: 'https://fit-s-marky.cz',
        tools: [
            'vercel',
            'nextjs',
            'react',
            'typescript',
            'lottie',
            'framer-motion',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Webové stránky instruktorky pilates a jógy.'},
            [LocalesEnum.en]: {description: 'Pilates and yoga instructor website.'},
        },
    },
    {
        id: 'dirtjob',
        title: 'Dirtjob',
        url: 'http://dirtjob.cz/',
        tools: [
            'octobercms',
            'vanillajs',
            'sass',
            'gulp',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Úklidová společnost s osobním přístupem. Úklid lze objednat z domova nebo kanceláře pomocí formuláře.'},
            [LocalesEnum.en]: {description: 'Cleaning company with a personal approach. Cleaning can be ordered from home or office using the form.'},
        },
    },
    {
        id: 'digitalvision',
        title: 'DigitalVision',
        url: 'https://digitalvision.cz/',
        tools: [
            'react',
            'lottie',
            'animejs',
            'postcss',
            'webpack',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Digitální agentura s širokým záběrem od designu, webového vývoje, přes tvorbu her až po marketing. Společnost zaměřující se na úzký vztah s klientem.'},
            [LocalesEnum.en]: {description: 'Digital agency with a wide range from design, web development, through game creation to marketing. A company focusing on a close relationship with the client.'},
        },
    },
    {
        id: 'stenclova',
        title: 'Ivana Štenclová',
        url: 'http://www.stenclova.cz',
        tools: [
            'octobercms',
            'vanillajs',
            'sass',
            'gulp',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Česká umělkyně, vyjadřující se převážně kresbou, ale také pomocí nástrojů jako je tavná pistole, laser nebo dekorační válečky.'},
            [LocalesEnum.en]: {description: 'Czech artist, expressing herself mostly by drawing, but also by tools such as a hot air gun, laser or decorative rollers.'},
        },
    },
    {
        id: 'avprague',
        title: 'AVPrague',
        url: 'https://avprague.in.progress',
        tools: [
            'octobercms',
            'jquery',
            'postcss',
            'gulp',
        ],
        locale: {
            [LocalesEnum.cs]: {description: 'Pronájem ubytování v Praze přes službu Airbnb.'},
            [LocalesEnum.en]: {description: 'Accommodation for rent in Prague via Airbnb.'},
        },
    },
];

export default projects;
