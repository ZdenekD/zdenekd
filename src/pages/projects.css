.article {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    position: relative;

    @media (--min-md) {
        align-items: center;
    }
}
