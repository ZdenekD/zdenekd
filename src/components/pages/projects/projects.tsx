import Head from 'next/head';
import {useRouter} from 'next/router';
import Layout from '@/components/layout';
import Heading from '@/components/layout/heading';
import Project from '@/components/project';
import pages from '@/data/pages';
import LocalesEnum from '@/enums/LocalesEnum';
import PagesEnum from '@/enums/PagesEnum';
import styles from './projects.module.css';

const Projects: React.FC = () => {
    const router = useRouter();
    const lang = router.locale as LocalesEnum || LocalesEnum.cs;
    const {description, heading} = pages[PagesEnum.projects].locale[lang];

    return (
        <>
            <Head>
                <meta name="description" content={description} />
                <title>{heading} | ZdenekD</title>
            </Head>
            <Layout className={styles.root}>
                <Heading data-testid="component-heading">{heading}</Heading>
                <article className={styles.article}>
                    <Project />
                </article>
            </Layout>
        </>
    );
};

export default Projects;
