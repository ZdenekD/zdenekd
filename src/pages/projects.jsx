import React from 'react';
import Main from '../components/layout/main';
import Project from '../components/project';
import data from '../data/pages';
import styles from './projects.css';

const Projects = () => (
    <Main data-test="component-main">
        <h2>{data.projects.heading}</h2>
        <article className={styles.article}>
            <Project />
        </article>
    </Main>
);

export default Projects;
