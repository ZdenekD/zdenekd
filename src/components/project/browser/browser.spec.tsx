import {shallow} from 'enzyme';
import findComponent from '@/__test__/utils/helpers';
import '@/mocks/__test__/nextRouterMock';
import {IProject} from '@/data/projects';
import Browser from '.';

const project: IProject = {
    id: 'id',
    title: 'title',
    locale: {cs: {description: 'description'}},
    url: 'http://url',
    tools: ['tool', 'tool'],
};

describe('Project/Browser', () => {
    it('renders without error', () => {
        const wrapper = shallow(
            <Browser
                isFirst={false}
                isLast={false}
                project={project}
            />
        );
        const component = findComponent(wrapper, 'component-browser');

        expect(component.exists()).toBe(true);
    });
});
