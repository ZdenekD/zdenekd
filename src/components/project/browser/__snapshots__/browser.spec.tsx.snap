// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser match snapshot 1`] = `
<section class="browser"
         data-test="component-browser"
>
  <header class="header">
    <span class="controls">
      <i class="control">
      </i>
      <i class="control">
      </i>
      <i class="control">
      </i>
    </span>
    <button type="button"
            class="prev "
            aria-label="Předchozí projekt"
            tabindex="0"
            data-test="component-prev"
    >
    </button>
    <button type="button"
            class="next "
            aria-label="Následující projekt"
            tabindex="0"
            data-test="component-next"
    >
    </button>
    <div class="addressbar http"
         data-test="component-addressbar"
    >
      <a href="http://url"
         target="_blank"
         rel="noreferrer noopener"
         aria-label="Otevřít referenci v novém okně"
      >
        http://url
      </a>
    </div>
    <div class="target">
      <a href="http://url"
         class="anchor targetLink"
         target="_blank"
         rel="noreferrer noopener"
         data-test="component-anchor"
      >
        <svg data-filename="src/assets/images/icon_open.svg">
        </svg>
      </a>
    </div>
  </header>
  <div class="wrapper"
       data-test="component-video"
       style="opacity:0"
  >
    <video playsinline
           autoplay
           muted
           loop
           preload="auto"
           class="video"
           title="Ukázka projektu: title. Stránky se nacházejí zde: http://url"
    >
      <track kind="captions">
      <track kind="description"
             label="title"
      >
      <source src="/video/id.mp4"
              type="video/mp4"
      >
    </video>
  </div>
</section>
`;
